shell/platform/android/io/flutter/embedding/FlutterView.java

```diff
@@ -38,1015 +33,781 @@
-        if (maskedAction == MotionEvent.ACTION_UP || maskedAction == MotionEvent.ACTION_POINTER_UP
-                || maskedAction == MotionEvent.ACTION_DOWN || maskedAction == MotionEvent.ACTION_POINTER_DOWN) {
-            addPointerForIndex(event, event.getActionIndex(), packet);
-        } else {
-            // ACTION_MOVE may not actually mean all pointers have moved
-            // but it's the responsibility of a later part of the system to
-            // ignore 0-deltas if desired.
-            for (int p = 0; p < pointerCount; p++) {
-                addPointerForIndex(event, p, packet);
-            }
-        }
-
-        assert packet.position() % (kPointerDataFieldCount * kBytePerField) == 0;
-        nativeDispatchPointerDataPacket(mNativeView.get(), packet, packet.position());
-        return true;
-    }
-
-    @Override
-    public boolean onHoverEvent(MotionEvent event) {
-        if (!isAttached()) {
-            return false;
-        }
-
-        boolean handled = handleAccessibilityHoverEvent(event);
-        if (!handled) {
-            // TODO(ianh): Expose hover events to the platform,
-            // implementing ADD, REMOVE, etc.
-        }
-        return handled;
-    }
-
+@TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+public class FlutterView extends SurfaceView implements
+    FlutterRenderer.RenderSurface,
+    AccessibilityManager.AccessibilityStateChangeListener {
+
+  private static final String TAG = "FlutterView";
+
+  // Must match the PointerChange enum in pointer.dart.
+  private static final int kPointerChangeCancel = 0;
+  private static final int kPointerChangeAdd = 1;
+  private static final int kPointerChangeRemove = 2;
+  private static final int kPointerChangeHover = 3;
+  private static final int kPointerChangeDown = 4;
+  private static final int kPointerChangeMove = 5;
+  private static final int kPointerChangeUp = 6;
+
+  // Must match the PointerDeviceKind enum in pointer.dart.
+  private static final int kPointerDeviceKindTouch = 0;
+  private static final int kPointerDeviceKindMouse = 1;
+  private static final int kPointerDeviceKindStylus = 2;
+  private static final int kPointerDeviceKindInvertedStylus = 3;
+  private static final int kPointerDeviceKindUnknown = 4;
+
+  static final class ViewportMetrics {
+    float devicePixelRatio = 1.0f;
+    int physicalWidth = 0;
+    int physicalHeight = 0;
+    int physicalPaddingTop = 0;
+    int physicalPaddingRight = 0;
+    int physicalPaddingBottom = 0;
+    int physicalPaddingLeft = 0;
+    int physicalViewInsetTop = 0;
+    int physicalViewInsetRight = 0;
+    int physicalViewInsetBottom = 0;
+    int physicalViewInsetLeft = 0;
+  }
+
+  private final InputMethodManager mImm;
+  private final ViewportMetrics mMetrics;
+  private final AccessibilityManager mAccessibilityManager;
+  private final AnimationScaleObserver mAnimationScaleObserver;
+
+  private FlutterRenderer flutterRenderer;
+  private BinaryMessenger pluginMessenger;
```

Comment:
nit: I believe the messenger not only used by plugins but also by "system channels" e.g the accessibility bridge.

Reply:
I guess it's a terminology thing, I personally don't think of things like the accessibility bridge or things like 'flutter/platform' as plugins (plugins sounds to me like something you can optionally include in your app)

Reply:
Interesting.  Maybe worth a discussion with some relevant people.  I thought part of Flutter's official philosophy was that "everything is a plugin."  I thought Flutter made the point publicly that "even core Flutter behavior in Android and iOS are actually just plugins."